statsmodels/base/_penalties.py

```diff
@@ -459,13 +444,10 @@ class ConstraintsPenalty(object):
 
     """
 
-    def __init__(self, penalty, weights=None, restriction=None):
+    def __init__(self, penalty, weights=1., restriction=None):
```

Comment:
I really think it would be better to leave weights=None for the default and then below use 

`self.weights = weights if weights is not None else 1.`

Same comment for other changes like this.

Then you can leave the doc string type for weight as `ndarray, optional`

Reply:
1. it is a misleading practice to use ```None``` as a default but always change it to ```1.0``` If it always changed to 1.0 it should be 1.0 by default.
2. as a user I couldn't understand the behavior of ```weights=None``` without looking into the code. It could be changed to 0.0,1.0,anystatisticsyouimagine under the hood and I now nothing about it without sources. Explicit is better than implicit.
3. Leaving ```weights=None``` required for this code to work ```weights = weights if weights is not None else wts``` but this approach leads us to the point where user could just add ```weights``` argument and not remove ```wts``` and we will pick ```weights``` and not force user to actually replace a deprecated argument.
@josef-pkt